src/borkdude/deps.clj

```diff
@@ -225,43 +225,172 @@ For more info, see:
+      (warn ::direct-download (.getMessage e))
+      false)))
+
+(def ^:private clojure-tools-info*
+  "A delay'd map with information about the clojure tools archive, where
+  to download it from, which files to extract and where to.
+
+  The map contains:
+
+  :ct-base-dir The relative top dir name to extract the archive files to.
+
+  :ct-error-exit-code The process exit code to return if the archive
+  cannot be downloaded.
+
+  :ct-aux-files-names Other important files in the archive.
+
+  :ct-jar-name The main clojure tools jar file in the archive.
+
+  :ct-url-str The url to download the archive from.
+
+  :ct-zip-name The file name to store the archive as."
+  (delay (let [version @version]
+           {:ct-base-dir  "ClojureTools"
+            :ct-error-exit-code 99
+            :ct-aux-files-names ["exec.jar" "example-deps.edn" "tools.edn"]
+            :ct-jar-name (format "clojure-tools-%s.jar" version)
+            :ct-url-str (format "https://download.clojure.org/install/clojure-tools-%s.zip" version)
+            :ct-zip-name "tools.zip"})))
 
 (defn unzip [zip-file destination-dir]
-  (let [zip-file (io/file zip-file)
+  (let [{:keys [ct-base-dir ct-aux-files-names ct-jar-name]} @clojure-tools-info*
+        zip-file (io/file zip-file)
         _ (.mkdirs (io/file destination-dir))
-        ^ClassLoader x nil
-        fs (FileSystems/newFileSystem (.toPath zip-file) x)]
-    (doseq [f [@clojure-tools-jar
-               "exec.jar"
-               "example-deps.edn"
-               "tools.edn"]]
-      (let [file-in-zip (.getPath fs "ClojureTools" (into-array String [f]))]
-        (Files/copy file-in-zip (.toPath (io/file destination-dir f))
-                    ^{:tag "[Ljava.nio.file.CopyOption;"}
-                    (into-array CopyOption
-                                [java.nio.file.StandardCopyOption/REPLACE_EXISTING]))))))
+        ^ClassLoader x nil]
+    (with-open [fs (FileSystems/newFileSystem (.toPath zip-file) x)]
+      (doseq [f (into [ct-jar-name] ct-aux-files-names)]
+        (let [file-in-zip (.getPath fs ct-base-dir (into-array String [f]))]
+          (Files/copy file-in-zip (.toPath (io/file destination-dir f))
+                      ^{:tag "[Ljava.nio.file.CopyOption;"}
+                      (into-array CopyOption
+                                  [java.nio.file.StandardCopyOption/REPLACE_EXISTING])))))))
+
+(defn- clojure-tools-java-downloader-spit
+  "Spits out and returns the path to `ClojureToolsDownloader.java` file
+  in DEST-DIR'ectory that when invoked (presumambly by the `java`
+  executable directly) with a source URL and destination file path in
+  args[0] and args[1] respectively, will download the source to
+  destination. No arguments validation is performed and returns exit
+  code 1 on failure."
+  [dest-dir]
+  (let [dest-file (.getCanonicalPath (io/file dest-dir "ClojureToolsDownloader.java"))]
+    (spit dest-file
+          (str "/** Auto-generated by " *file* ". **/"
```

Comment:
We can format this program by doing:

```
(str "<newline>




" ;; end of program
```

I it is more convenient to have it as a single string, so when we make changes, you don't have to re-format any other lines.